tests/test_model_saving.py

```diff
@@ -112,6 +113,30 @@ def test_functional_model_saving():
     assert_allclose(out, out2, atol=1e-05)
 
 
+def test_functional_model_saving_with_sparse():
+    inputs = Input(shape=(3,), sparse=True)
+    x = Dense(2)(inputs)
+    outputs = Dense(3)(x)
+
+    model = Model(inputs, outputs)
+    model.compile(loss=losses.MSE,
+                  optimizer=optimizers.Adam(),
+                  metrics=[metrics.categorical_accuracy])
+    x = rand(1, 3, density=0.2, format='csr')
+    y = rand(1, 3, density=0.2, format='csr')
```

Comment:
labels are not sparse, can we use normal dense labels

Reply:
done